zavod/zavod/shed/gpt.py

```diff
@@ -120,3 +170,48 @@ def run_text_prompt(
     data = json.loads(response.choices[0].message.content)
     context.cache.set_json(cache_key, data)
     return data
+
+
+def run_typed_text_prompt(
+    context: Context,
+    prompt: str,
+    string: str,
+    response_type: Type[ResponseType],
+    max_tokens: int = 3000,
+    cache_days: int = 100,
+    model: str = "gpt-4o",
+) -> ResponseType:
+    """Run a text prompt."""
+    client = get_client()
+    cache_hash = sha1(string.encode("utf-8"))
+    cache_hash.update(prompt.encode("utf-8"))
+    json_schema = response_type.model_json_schema()
+    cache_hash.update(json.dumps(json_schema, sort_keys=True).encode("utf-8"))
+    cache_key = cache_hash.hexdigest()
+    cached_data = context.cache.get_json(cache_key, max_age=cache_days)
+    if cached_data is not None:
+        log.info("GPT cache hit: %s" % string[:50])
+        return response_type.model_validate(cached_data)
+    log.info("Prompting %r for: %s" % (model, string[:50]))
+    response = client.chat.completions.parse(
```

Comment:
Had to switch this to `client.beta.chat.completions.parse` to get it to work for some reason?

Reply:
Yeah, I forgot the upgrade eager. I upped the version.